### PR TITLE
remove glob-base dependency

### PR DESCRIPTION
**Pull request by lkuechler**
Issue: #15174 

## What I did
I removed the glob-base dependency to resolve the following security vulnerability #15174 

## How to test
No new tests should be needed.

## Addition information
I had to reimplement some logic from glob-base. The returned `glob` is for example filled with the input path if no glob was detected.

### CODE REVIEW

**Comment by shilman** [comment]
@lkuechler looks like the build is failing in examples:

```
ERR! => Failed to build the preview
ERR! Module not found: Error: Can't resolve 'src' in '/tmp/storybook/examples/react-ts'
ERR! ModuleNotFoundError: Module not found: Error: Can't resolve 'src' in '/tmp/storybook/examples/react-ts'
ERR!     at /tmp/storybook/node_modules/webpack/lib/Compilation.js:925:10
ERR!     at /tmp/storybook/node_modules/webpack/lib/ContextModuleFactory.js:143:23
ERR!     at /tmp/storybook/node_modules/neo-async/async.js:2830:7
ERR!     at /tmp/storybook/node_modules/neo-async/async.js:6877:13
ERR!     at /tmp/storybook/node_modules/webpack/lib/ContextModuleFactory.js:118:26
ERR!     at /tmp/storybook/node_modules/enhanced-resolve/lib/Resolver.js:213:14
ERR!     at /tmp/storybook/node_modules/enhanced-resolve/lib/Resolver.js:285:5
ERR!     at eval (eval at create (/tmp/storybook/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
ERR!     at /tmp/storybook/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7
ERR!     at /tmp/storybook/node_modules/enhanced-resolve/lib/Resolver.js:285:5
ERR!     at eval (eval at create (/tmp/storybook/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:13:1)
ERR!     at /tmp/storybook/node_modules/enhanced-resolve/lib/Resolver.js:285:5
ERR!     at eval (eval at create (/tmp/storybook/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:25:1)
ERR!     at /tmp/storybook/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
ERR!     at /tmp/storybook/node_modules/enhanced-resolve/lib/Resolver.js:285:5
ERR!     at eval (eval at create (/tmp/storybook/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:41:1)
undefined
```

**Comment by lkuechler** [comment]
hi @shilman,
thanks for the hint. I have now fixed the errors and the only pipelines that now fail also seem to fail in all other pull-requests :)

**Comment by powerchelle** [comment]
Hi @lkuechler, curious if this is still being actively pursued. Looks like last update was beginning of July. Do you have an ETA? Thanks in advance!

File: lib/core-common/src/utils/to-require-context.ts [hunk 1]
```diff
@@ -25,21 +25,33 @@ const detectBadGlob = (val: string) => {
 const isObject = (val: Record<string, any>) =>
   val != null && typeof val === 'object' && Array.isArray(val) === false;
 
+const dirname = (pattern: string) => {
+  if (pattern.slice(-1) === '/') return pattern;
```
Comment (shilman) [inline lib/core-common/src/utils/to-require-context.ts:29]: what if the user passes a directory but doesn't include a trailing slash? any chance you can provide a test case for this and for any other cases you considered in this logic?

**Review by shilman — CHANGES_REQUESTED** [review]

File: lib/core-common/src/utils/to-require-context.ts (hunk 1 above)
Comment (lkuechler) [reply lib/core-common/src/utils/to-require-context.ts]: This was behaviour was something that I saw in the glob-base dependency that was behaving a bit strange. I have now checked with the implementation in storybook on how it currently behaves and I believe that this extra logic did not make any difference. I have removed this now.

I think the current test are already very good but I added two more that focus on the folder pattern.

**Review by shilman — APPROVED** [review]
Thanks for the follow-through @lkuechler -- looking great! 😍

**Pull request merged**